src/client/Item/Item.ts

```diff
@@ -4,17 +4,50 @@ import { RequestOptions, Response } from "../../request";
 import { Container } from "../Container";
 import { ItemResponse } from "./ItemResponse";
 
+/**
+ * Used to perform operations on a specific item.
+ *
+ * @see {@link Items} for operations on all items; see `container.items`.
+ */
 export class Item {
   private client: CosmosClient;
+  /**
+   * Returns a reference URL to the resource. Used for linking in Permissions.
+   */
   public get url() {
     return UriFactory.createDocumentUri(this.container.database.id, this.container.id, this.id);
   }
 
+  /**
+   * @hidden
+   * @param container The parent {@link Container}.
+   * @param id The id of the given {@link Item}.
+   * @param primaryKey The primary key of the given {@link Item} (only for partitioned containers).
+   */
   constructor(public readonly container: Container, public readonly id: string, public readonly primaryKey: string) {
     this.client = this.container.database.client;
   }
 
+  /**
+   * Read the item's definition.
+   *
+   * There is no set schema for JSON items. They may contain any number of custom properties.
+   *
+   * @param options Additional options for the request, such as the partition key.
+   * Note, if you provide a partition key here, it will override the primary key on this object.
```

Comment:
Can you be more specific on this line? Kinda sounds like a new primary key will be written?